### PR TITLE
Get user by `uid` to fetch information of ProfileScreen

### PR DESCRIPTION
**Pull request by barmettlerl**

### CODE REVIEW

File: mobile/src/main/java/ch/epfl/sdp/mobile/application/social/SocialFacade.kt [hunk 1]
```diff
@@ -40,20 +41,19 @@ class SocialFacade(private val auth: Auth, private val store: Store) {
...
+   * @param uid search criteria.
    * @param user the [AuthenticationUser] that is performing the get.
    */
   fun get(
-      name: String,
+      uid: String,
       user: AuthenticationUser = NotAuthenticatedUser,
-  ): Flow<Profile> {
+  ): Flow<Profile?> {
+    Log.i("info", uid)
```
Comment (alexandrepiveteau) [inline mobile/src/main/java/ch/epfl/sdp/mobile/application/social/SocialFacade.kt:51]:
```suggestion
```

File: mobile/src/main/java/ch/epfl/sdp/mobile/application/social/SocialFacade.kt [hunk 2]
```diff
@@ -40,20 +41,19 @@ class SocialFacade(private val auth: Auth, private val store: Store) {
...
         .collection("users")
-        .whereEquals("name", name)
-        .limit(1)
+        .document(uid)
         .asFlow<ProfileDocument>()
-        .map { it.mapNotNull { doc -> doc?.toProfile(user) } }
-        .map { it.first() }
-        .catch { null }
+        .map { doc -> doc?.toProfile(user) }
+        .catch { emit(null) }
```
Comment (alexandrepiveteau) [inline mobile/src/main/java/ch/epfl/sdp/mobile/application/social/SocialFacade.kt:57]: Question : is this `catch { ... }` needed ?

File: mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulFollowingScreen.kt [hunk 3]
```diff
@@ -95,7 +97,7 @@ private class SnapshotSocialScreenState(
     override var mode: SocialScreenState.Mode,
     override val searchFieldInteraction: MutableInteractionSource,
     private val scope: CoroutineScope,
-    private val onPersonClick: (Person) -> Unit
+    private val onPersonElementClick: (Person) -> Unit
```
Comment (alexandrepiveteau) [inline mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulFollowingScreen.kt:100]: You should make this a `(ProfileAdapter) -> Unit`, so you don't have to add `uid` to `Person`.

Comment (alexandrepiveteau) [inline mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulFollowingScreen.kt:100]: The name of the callback should maybe be `onPersonItemClick`. In Compose, elements of a list are called items.

File: mobile/src/main/java/ch/epfl/sdp/mobile/application/social/SocialFacade.kt [hunk 4]
```diff
@@ -40,20 +41,19 @@ class SocialFacade(private val auth: Auth, private val store: Store) {
   /**
    * Returns a [Flow] of the [Profile] by the given name
    *
-   * @param name text search criteria.
+   * @param uid search criteria.
    * @param user the [AuthenticationUser] that is performing the get.
    */
   fun get(
```
Comment (alexandrepiveteau) [inline mobile/src/main/java/ch/epfl/sdp/mobile/application/social/SocialFacade.kt:47]: Already in another PR's comments I think, but this should maybe be called `profile(...)`, since `get` is very generic, especially in a `SocialFacade`.

File: mobile/src/main/java/ch/epfl/sdp/mobile/ui/social/Person.kt [hunk 5]
```diff
@@ -5,6 +5,9 @@ import ch.epfl.sdp.mobile.application.Profile.Color
 /** An interface representing a list of people you're following. */
 interface Person {
 
+  /** The the unique identifier of the Person. */
+  val uid: String
```
Comment (alexandrepiveteau) [inline mobile/src/main/java/ch/epfl/sdp/mobile/ui/social/Person.kt:9]: You must not add a `uid` to `Person`. `Person` is a view interface, whereas the `uid` belongs to the model.

File: mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/application/SocialFacadeTest.kt [hunk 6]
```diff
@@ -109,8 +109,8 @@ class SocialFacadeTest {
     }
     val facade = SocialFacade(auth, store)
 
-    val user = facade.get("test").first()
-    Truth.assertThat(user.name).isEqualTo("test")
+    val user = facade.get("test").firstOrNull()
```
Comment (alexandrepiveteau) [inline mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/application/SocialFacadeTest.kt:112]: Does this test actually pass ? I expect the `facade` not to find any document with the `uid` set to `test`, and the assertion to fail.

File: mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulHome.kt [hunk 7]
```diff
@@ -54,7 +54,7 @@ fun StatefulHome(
   val section = entry?.toSection() ?: HomeSection.Social
 
   val onPersonClick: (person: Person) -> Unit = { person ->
-    controller.navigate("$ProfileRoute/${person.name}")
+    controller.navigate("$ProfileRoute/${person.uid}")
   }
```
Comment (alexandrepiveteau) [inline mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulHome.kt:58]: To get the `uid`, use a `ProfileAdapter` here.

**Review by alexandrepiveteau — CHANGES_REQUESTED** [review]

File: mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/application/SocialFacadeTest.kt (hunk 6 above)
Comment (barmettlerl) [reply mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/application/SocialFacadeTest.kt:112]: It does, because I catch the error and return `null`. But I see, This is not a good approach

File: mobile/src/main/java/ch/epfl/sdp/mobile/application/social/SocialFacade.kt [hunk 8]
```diff
@@ -40,20 +39,15 @@ class SocialFacade(private val auth: Auth, private val store: Store) {
   /**
    * Returns a [Flow] of the [Profile] by the given name
    *
-   * @param name text search criteria.
+   * @param uid search criteria.
```
Comment (alexandrepiveteau) [inline mobile/src/main/java/ch/epfl/sdp/mobile/application/social/SocialFacade.kt:42]:
```suggestion
   * @param uid the unique identifiers of the profile.
```

File: mobile/src/main/java/ch/epfl/sdp/mobile/application/social/SocialFacade.kt [hunk 9]
```diff
@@ -40,20 +39,15 @@ class SocialFacade(private val auth: Auth, private val store: Store) {
   /**
    * Returns a [Flow] of the [Profile] by the given name
```
Comment (alexandrepiveteau) [inline mobile/src/main/java/ch/epfl/sdp/mobile/application/social/SocialFacade.kt:40]:
```suggestion
   * Returns a [Flow] of the [Profile] corresponding to a given unique identifier.
```

File: mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulFollowingScreen.kt [hunk 10]
```diff
@@ -86,6 +87,7 @@ fun StatefulFollowingScreen(
  * @param mode the current [SocialScreenState.Mode] of the social screen.
  * @param searchFieldInteraction the [MutableInteractionSource] of the search field.
  * @param scope the [CoroutineScope] on which requests are performed.
+ * @param onPersonElementClick on click of person list element
```
Comment (alexandrepiveteau) [inline mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulFollowingScreen.kt:90]:
```suggestion
 * @param onPersonItemClick on click of person list element
```

**Review by alexandrepiveteau — COMMENTED** [review]
Some small adjustments to do regarding the documentation which you should be able to apply as-is.

File: mobile/src/main/java/ch/epfl/sdp/mobile/ui/profile/UserScreen.kt [hunk 11]
```diff
@@ -32,7 +32,7 @@ fun UserScreen(
     profileTabBar: @Composable () -> Unit,
     matches: List<ChessMatch>,
     lazyColumnState: LazyListState,
-    modifier: Modifier = Modifier,
+    modifier: Modifier,
```
Comment (alexandrepiveteau) [inline mobile/src/main/java/ch/epfl/sdp/mobile/ui/profile/UserScreen.kt:35]: Question : why did you remove this default `Modifier` ?

Comment (barmettlerl) [reply mobile/src/main/java/ch/epfl/sdp/mobile/ui/profile/UserScreen.kt:35]: Not needed and to improve test coverage

**Review by alexandrepiveteau — APPROVED** [review]

**Review by KurohanaJuri — DISMISSED** [review]
For the next time, can you add a little summary about PR that help people to understand the reason of this kind of changes :)

**Review by alexandrepiveteau — APPROVED** [review]
